### PR TITLE
infra: Parse results to HTML

### PR DESCRIPTION
**Pull request by sshnaidm**
parse results from Junit XML to HTML to see the results in more human-friendly way.

### CODE REVIEW

**Comment by sshnaidm** [comment]
Example: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-kni_cnf-features-deploy/1750/pull-ci-openshift-kni-cnf-features-deploy-master-e2e-aws-ci-tests/1744012903003983872/artifacts/e2e-aws-ci-tests/e2e-aws-ci-tests/artifacts/cnftests.html

**Comment by sshnaidm** [comment]
> @sshnaidm Thanks for putting this together! I think that it could be beneficial to add a link to the artifacts of the job, (like we have in the `upstream-ci-reporter`) because this is always the next step when I want to dive into troubleshooting a failed test case.

Actually you can't go to this report without entering artifacts, it's not mail report, just junit to html converter. In time it's being converted you don't know yet the artifact link, it's still job running.

**Comment by sshnaidm** [comment]
**All passed results:** https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-kni_cnf-features-deploy/1750/pull-ci-openshift-kni-cnf-features-deploy-master-e2e-aws-ci-tests/1745382127916027904/artifacts/e2e-aws-ci-tests/e2e-aws-ci-tests/artifacts/cnftests.html 
**Validations:** https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-kni_cnf-features-deploy/1750/pull-ci-openshift-kni-cnf-features-deploy-master-e2e-aws-ci-tests/1745382127916027904/artifacts/e2e-aws-ci-tests/e2e-aws-ci-tests/artifacts/validation.html
**Config suite**: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-kni_cnf-features-deploy/1750/pull-ci-openshift-kni-cnf-features-deploy-master-e2e-aws-ci-tests/1745382127916027904/artifacts/e2e-aws-ci-tests/e2e-aws-ci-tests/artifacts/setup.html

File: hack/run-functests.sh [hunk 1]
```diff
@@ -69,7 +69,9 @@ fi
 if [ ! -f "cnf-tests/bin/junit-merger" ]; then
   go build -o cnf-tests/bin/junit-merger cnf-tests/testsuites/pkg/junit-merger/junit-merger.go
 fi
-
+if [ ! -f "cnf-tests/bin/j2html" ]; then
+    go build -o cnf-tests/bin/j2html cnf-tests/testsuites/pkg/j2html/j2html.go
+fi
```
[the remark this answers is not in the document]
Comment (sshnaidm) [reply hack/run-functests.sh]: done

File: hack/run-functests.sh [hunk 2]
```diff
@@ -69,7 +69,9 @@ fi
 if [ ! -f "cnf-tests/bin/junit-merger" ]; then
   go build -o cnf-tests/bin/junit-merger cnf-tests/testsuites/pkg/junit-merger/junit-merger.go
 fi
-
+if [ ! -f "cnf-tests/bin/j2html" ] && [ "$JUNIT_TO_HTML" == "true" ]; then
+    go build -o cnf-tests/bin/j2html cnf-tests/testsuites/pkg/j2html/j2html.go
+fi
```
Comment (liornoy) [inline hack/run-functests.sh:74]: I mean moving this below line 144

Comment (sshnaidm) [reply hack/run-functests.sh:74]: I think it's better to keep all building code in one place consolidated

Comment (liornoy) [reply hack/run-functests.sh:74]: Ok. I agree that this choice will benefit more in improving readability then saving us from building this bin which is small anyway.

Comment (liornoy) [reply hack/run-functests.sh:74]: it's lgtm too.

**Pull request merged**